libs/ml/include/ml/dataloaders/commodity_dataloader.hpp

```diff
@@ -29,16 +29,35 @@ namespace fetch {
 namespace ml {
 namespace dataloaders {
 
-inline std::pair<math::SizeType, math::SizeType> count_rows_cols(std::string const &filename)
+/**
+ * Loads a csv file into an ArrayType (a Tensor)
+ * The Tensor will have the same number of rows as this file has (minus rows_to_skip) and the same
+ * number of columns (minus cols_to_skip) as the file, unless transpose=true is specified in which
+ * case it will be transposed.
+ * @param filename  name of the file
+ * @param cols_to_skip  number of columns to skip
+ * @param rows_to_skip  number of rows to skip
+ * @param transpose  whether to transpose the resulting Tensor
+ * @return ArrayType with data
+ */
+template <typename ArrayType>
+ArrayType ReadCSV(std::string const &filename, math::SizeType const cols_to_skip = 0,
```

Comment:
please move this to ml/dataloaders/ReadCSV.hpp
since this is a standalone utility